### PR TITLE
Use variable typing in comments to support python 3.5

### PR DESCRIPTION
**Pull request by dhood**
Fixes https://github.com/ros2/launch/issues/78

This fixes the build on xenial.

@wjwwood there are a few lines that are too long (https://ci.ros2.org/job/ci_linux/4703/testReport/) but I haven't got mypy setup to check what wrapping will be supported. What invocation do you use? I get `Can't find package 'launch'`  when calling `python3 -m mypy --package launch`...

### CODE REVIEW

**Comment by wjwwood** [comment]
Not sure let me check my integration into my editor.

**Comment by wjwwood** [comment]
I'm using `--ignore-missing-imports`.

File: launch/launch/actions/execute_process.py [hunk 1]
```diff
@@ -71,15 +71,15 @@ class ExecuteProcess(Action):
...
         self, *,
-        cmd: Iterable[SomeSubstitutionsType],
-        cwd: Optional[SomeSubstitutionsType] = None,
-        env: Optional[Dict[SomeSubstitutionsType, SomeSubstitutionsType]] = None,
-        shell: bool = False,
-        sigterm_timeout: SomeSubstitutionsType = LaunchConfiguration('sigterm_timeout', default=5),
-        sigkill_timeout: SomeSubstitutionsType = LaunchConfiguration('sigkill_timeout', default=5),
-        prefix: Optional[SomeSubstitutionsType] = None,
-        output: Optional[Text] = None,
-        log_cmd: bool = False,
```
Comment (wjwwood) [inline launch/launch/actions/execute_process.py]: I thought these kinds worked in Python3.5?

Comment (dhood) [reply launch/launch/actions/execute_process.py]: oh, maybe I went overkill. I'll check this afternoon.

**Comment by dhood** [comment]
OK @wjwwood you were right and I had more changes than necessary before. It's been reduced to the minimal diff, which does include some code changes, if you could check them thanks (the changes are in individual commits).

File: launch/launch/launch_introspector.py [hunk 2]
```diff
@@ -111,24 +114,26 @@ def format_event_handler(event_handler: EventHandler) -> List[Text]:
...
-                 for k, v in action.env.items()] + '}'),
-            action.shell,
+                 for k, v in typed_action.env.items()] + '}'),
+            typed_action.shell,
         )
         return [msg]
     elif is_a(action, RegisterEventHandler):
-        result = ["RegisterEventHandler('{}'):".format(action.event_handler)]
-        result.extend(indent(format_event_handler(action.event_handler)))
+        typed_action2 = cast(RegisterEventHandler, action)
```
Comment (dhood) [inline launch/launch/launch_introspector.py]: mypy wasn't happy with reuse of the same variable name as above (even though they're in different branches)...

File: launch/launch/launch_introspector.py [hunk 3]
```diff
@@ -100,7 +103,7 @@ def format_event_handler(event_handler: EventHandler) -> List[Text]:
     """Return a text representation of an event handler."""
     if hasattr(event_handler, 'describe'):
         # TODO(wjwwood): consider supporting mode complex descriptions of branching
-        description, entities = event_handler.describe()
+        description, entities = event_handler.describe()  # type: ignore
```
Comment (dhood) [inline launch/launch/launch_introspector.py]: mypy was complaining about that `EventHandler` doesn't have a `describe` method: I think that ignoring it is the appropriate thing to do here.

**Comment by dhood** [comment]
xenial [Build Status](https://ci.ros2.org/view/nightly/job/ci_linux/4716/)
bionic [Build Status](https://ci.ros2.org/job/ci_linux/4717/)

File: launch/launch/actions/timer_action.py [hunk 4]
```diff
@@ -55,7 +55,7 @@ class TimerAction(Action):
     def __init__(
         self, *,
         period: Union[float, SomeSubstitutionsType],
-        actions: Iterable[LaunchDescriptionEntity],
+        actions: Iterable[LaunchDescriptionEntity]
```
[the remark this answers is not in the document]
Comment (dhood) [reply launch/launch/actions/timer_action.py]: Yeah, I know that we do it sometimes intentionally (thanks for the explanation though :) ) but it was causing syntax errors with coverage:
```
launch/actions/execute_process.py:37: in <module>
    from .timer_action import TimerAction
E     File "/home/dhood/ros2_ws/src/ros2/launch/launch/launch/actions/timer_action.py", line 59
E       ) -> None:
E       ^
E   SyntaxError: invalid syntax
```

(presumably something unique to `__init__`s)

[the remark this answers is not in the document]
Comment (dhood) [reply launch/launch/actions/timer_action.py]: this looks like the python ticket for the curious! (fixed in 3.6) https://bugs.python.org/issue9232

[the remark this answers is not in the document]
Comment (wjwwood) [reply launch/launch/actions/timer_action.py]: Hmm, weird. :)

**Pull request merged**